Content.Server/Imperial/Radio/JobPlayerAndColor.cs

```diff
@@ -0,0 +1,173 @@
+using System.Text.RegularExpressions;
+using Content.Server.Access.Systems;
+using Robust.Shared.Utility;
+using Robust.Shared.Prototypes;
+using Content.Shared.Imperial.Radio;
+
+namespace Content.Server.Imperial.Radio;
+
+/// <summary>
+/// Данный класс отвечает за отображения должностей в канале рации. То есть, если ошибка в Content.Radi.EnitySystem - вам сюда. Так же в самом оригинальном коде рации есть способ быстро востановить его оригинальную работу без моего класса
+/// </summary>
+public sealed class JobPlayerAndColor : EntitySystem
+{
+    // Для работы с id картой
+    [Dependency] private readonly IdCardSystem _idCardSystem = default!;
+    [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+
+    // Стреляй, О(1) не отдам
+    private static Dictionary<string, string> SaveColorForJob = new Dictionary<string, string>();
+
+    // Метод, который проверяет нужно ли заполнять словарь
+    public void IsDictionaryEmpty()
+    {
+        // Проверяем, пуст ли словарь, если пустой — инициализируем
+        if (SaveColorForJob.Count == 0)
+        {
+            LoadPrototypes();
+        }
+    }
+
+    // Метод, который заполняет словарь. Вызывается только через IdColorService
+    public void LoadPrototypes()
+    {
+        // Получаем все прототипы для idColor
+        var prototypes = IoCManager.Resolve<IPrototypeManager>().EnumeratePrototypes<IdColorAndJobPrototype>();
+
+        foreach (var proto in prototypes)
+        {
+            // Добавляем в словарь только уникальные ID
+            if (!SaveColorForJob.ContainsKey(proto.ID))
+            {
+                SaveColorForJob.Add(proto.ID, proto.Color);
+            }
+        }
+    }
+
+    // Словарь для перевода должностей и для должностей состоящих не только из букв
+    private static readonly Dictionary<string, string> JobTranslations = new()
+    {
+        { "Central Commander", "Центральное Командование" },
+        { "Centcom Quarantine Officer", "РХБЗЗ" }
+    };
+
+    private string TranslateJob(string job)
+    {
+        // Ищем перевод в словаре
+        if (JobTranslations.TryGetValue(job, out var translatedJob))
+        {
+            return translatedJob;
+        }
+
+        // Возвращаем оригинал, если перевода нет
+        return job;
+    }
+
+	/// <summary>
+    /// Для EnityUid ищется карта в руках или в пда. Если карта нашлась, мы смотрим какая там работа и достаём её
+    /// </summary>
+    /// <param name="uid"> uid игрока, у которого мы ищем карту</param>
+    /// <returns></returns>
+    public string? GetJobPlayer(EntityUid uid)
```

Comment:
Нуллейбл и так не возвращается
```suggestion
    public string GetJobPlayer(EntityUid uid)
```

Reply:
Опять же вызов критической ошибки. Выше объяснялось. Закрываю с пометкой ### "НИ В КОЕМ СЛУЧАЕ ТАК НЕ ДЕЛАТЬ!"